cedar-lean/Cedar/Thm/Authorization/Evaluator.lean

```diff
@@ -113,8 +113,8 @@ theorem ways_and_can_error {e₁ e₂ : Expr} {request : Request} {entities : En
       case bool b =>
         cases b with
         | true =>
-          simp
-          simp [h_e₁] at h₁
+          simp only [true_and]
+          simp only [h_e₁] at h₁
```

Comment:
in this (and other) cases, would it be clearer to replace `simp only` with just `rw`?  What are the differences?  I think one difference is that `rw` would make you be explicit about which direction to rewrite (requires `\l` in some cases); we can decide whether we like that

Reply:
`simp` does quite a bit more than `rw` (https://leanprover.github.io/theorem_proving_in_lean4/tactics.html#rewriting). `rw` fails on both of these. Btw, `simp` also accepts the `\l` syntax.

Reply:
Thanks for making this PR!  I've been going through some other proofs and implementing the "no non-terminal simps rule", and haven't gotten this far yet :)